.pre-commit-config.yaml

```diff
@@ -0,0 +1,12 @@
+fail_fast: false
+repos:
+  - repo: https://github.com/pocc/pre-commit-hooks
+    rev: master
+    hooks:
+      - id: clang-format
```

Comment:
Do these hooks only run if you're changing files relevant to the hook? It would be nice to not require you to install clang-format if you're only changing docs files and not require node if you're only changing app files.

Reply:
The pre-commit hooks are only run on the files that are changed. However, an 'environment' is created for all hooks that are specified, whether they are needed for the changed files or not. For the clang-format hook, clang-format needs to be installed in the system beforehand. 

pre-commit hooks are optional, so anyone who does not want to use them can just decide not to run `pre-commit install` and not be bothered with them.